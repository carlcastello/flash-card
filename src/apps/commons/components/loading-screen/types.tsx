export interface IOwnProps {
  classes: {
    boxContainer: string,
    boxContent: string,
    typography: string,
    linearProgressBar: string,
  }
}