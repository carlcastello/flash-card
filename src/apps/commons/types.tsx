export enum QuestionType {
  WORD,
  QUESTIONAIRE 
}

export interface IFlashcard {
  id: string,
  question: string,
  subQuestion?: string,
  hint?: string,
  questionType: QuestionType,
  answer: string,
}

export interface IQuiz {  
  id: string,
  quizSummary: IQuizSummary,
  quizQuestions: {
    flashcards: IFlashcard[]
  }
}

export interface IQuizSummary {
  title: string,
  description: string
} 

export interface IQuizSummaryCard {
  id: string,
  quizSummary: IQuizSummary
}

export interface IQuizQuestion {
  question: string,
  subQuestion?: string,
  hint?: string,
  answer: string,
}