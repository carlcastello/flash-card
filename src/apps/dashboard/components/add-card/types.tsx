export interface IOwnProps {
  onClick: () => void,
  hasHoverEffect?: boolean,
  classes: {
    paperContainer: string,
    boxContainer: string,
  }
}

export interface IOwnState {
  paperElevation: number
}