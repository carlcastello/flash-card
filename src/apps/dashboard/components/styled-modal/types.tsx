export interface IOwnProps {
  isOpen: boolean,
  onConfirm?: () => void,
  onIgnore: () => void,
  isLoading?: boolean,

  classes: {
    modalContainer: string,
    modalPaperContainer: string,
    closeButton: string,
  }
}