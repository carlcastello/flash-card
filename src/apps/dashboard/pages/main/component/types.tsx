import { IQuizSummaryCard } from "../../../../commons/types";
import { RouteComponentProps } from "react-router-dom";


export interface IOwnProps extends RouteComponentProps<any> {
  isFullPageLoading: boolean,
  requiredData: string[],
  createdQuizes: IQuizSummaryCard[],
  fetchCreatedQuizes: () => Promise<any>,
  deleteCreatedQuiz: (quizId: string) => Promise<any>,
  isDeletingQuiz: boolean,
  classes: {
    gridContainer: string,
  }
};

export interface IOwnState {
  toggleModuleId: string,
  moduleModalOpen: boolean,
};