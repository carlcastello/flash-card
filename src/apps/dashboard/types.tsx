export interface IOwnProps {
  classes: {
    boxContainer: string,
  }
}