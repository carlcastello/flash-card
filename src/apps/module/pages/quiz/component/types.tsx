import { FlashcardStatus } from '../../../components/flash-card/types';
import { RouteComponentProps } from 'react-router-dom';
import { IQuestionCard } from '../../../../commons/types';

import { QuizStatus } from './enum';

export interface IOwnProps extends RouteComponentProps<any> {
  questions: IQuestionCard[],
  requiredData: string[],

  fetchQuiz: (quizId: string) => void,

  classes: {
    boxContent: string,
    boxContainer: string,
    progressBarContainer: string,
    iconButton: string,
  }
}

export interface IOwnState {
  questionIndex: number,
  progressIndex: number,
  currentFlashcardStatus: FlashcardStatus,
  quizStatus: QuizStatus,
  confirmModalToogle: boolean
}