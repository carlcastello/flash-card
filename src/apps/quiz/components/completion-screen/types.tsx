export interface IOwnProps {
  onClose: () => void,
  onReview: () => void,
  classes: {
    
  }
}