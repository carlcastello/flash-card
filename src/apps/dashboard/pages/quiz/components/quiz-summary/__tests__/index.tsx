import React from "react";

import { shallow } from "enzyme";

import { QuizSummary } from "../index";
import toJson from "enzyme-to-json";
import Form from "../../../../../components/form";
import { Box } from "@material-ui/core";
import { IFormResponse } from "../../../../../components/form/types";

describe("The Quiz Summary", () => {

  let wrapper: any;
  let instance: any;

  let onCreateQuizSummaryFunction: any;
  let onSaveQuizSummaryFunction: any;

  beforeEach(() => {
    onCreateQuizSummaryFunction = jest.fn();
    onSaveQuizSummaryFunction = jest.fn();

    wrapper = shallow(
      <QuizSummary
        isLoading={true}
        quizId={undefined}
        quizSummary={undefined}
        onCreateQuizSummary={onCreateQuizSummaryFunction}
        onSaveQuizSummary={onSaveQuizSummaryFunction}
        />
    );

    instance = wrapper.instance();
  })

  describe("The render function", () => {
    it("renders the form container", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Box)).toHaveLength(2);
      expect(wrapper.find(Form)).toHaveLength(1);
    });
  });

  describe("The onSubmit functions", () => {
    it("doesn't have a quizId thus call onCreateQuizSummary", () => {
      const formResponse: IFormResponse = { 
        title: "potato",
        description: "salad",
        hello: "world"
      }; 
      wrapper.setProps({ quizId: undefined });
      instance.onFormSubmit(formResponse);
      expect(onCreateQuizSummaryFunction).toHaveBeenCalledWith(formResponse);
    });

    it("does have a quizId thus call onSaveQuizSummary", () => {
      const quizId: string = "this-is-and-id";
      const formResponse: IFormResponse = {
        title: "potato",
        description: "salad",
        hello: "world"
      }; 
      wrapper.setProps({ quizId });
      instance.onFormSubmit(formResponse);
      expect(onSaveQuizSummaryFunction).toHaveBeenCalledWith(quizId, formResponse);

    });
  })
});