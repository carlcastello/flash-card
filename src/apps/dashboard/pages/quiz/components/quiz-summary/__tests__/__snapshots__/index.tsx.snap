// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Quiz Summary The render function renders the form container 1`] = `
<WithStyles(ForwardRef(Paper))
  elevation={3}
>
  <Styled(MuiBox)
    p={5}
  >
    <Styled(MuiBox)
      pb={2}
    >
      <WithStyles(ForwardRef(Typography))
        variant="h4"
      >
        Quiz Summary
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
    <WithStyles(Form)
      fields={
        Array [
          Object {
            "id": "title",
            "label": "Title",
            "required": true,
            "type": "text",
            "validator": [Function],
            "value": "",
          },
          Object {
            "id": "description",
            "label": "Description",
            "type": "text",
            "validator": [Function],
            "value": "",
          },
        ]
      }
      isLoading={true}
      onSuccess={[Function]}
    />
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Paper))>
`;
